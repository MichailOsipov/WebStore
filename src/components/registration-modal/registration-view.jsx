import * as React from 'react';
import Button from '@material-ui/core/Button';
import {RegistrationModalForm} from './registration-modal-form';

export const RegistrationView = ({onRegister, onGoMainPage}) => (
    <div>
        Регистрация
        <RegistrationModalForm />
        <Button variant="contained" onClick={onRegister}>Зарегистрироваться</Button>
        <Button onClick={onGoMainPage}>Назад</Button>
    </div>
);
