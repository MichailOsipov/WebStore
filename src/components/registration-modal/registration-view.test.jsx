import * as React from 'react';
import {shallow} from 'enzyme';
import {RegistrationView} from './registration-view';

describe('RegistrationView', () => {
    test('rendering', () => {
        const wrapper = shallow(
            <RegistrationView
                onRegister={jest.fn()}
                onGoMainPage={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
