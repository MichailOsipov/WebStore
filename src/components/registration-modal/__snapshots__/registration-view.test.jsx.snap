// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationView rendering 1`] = `
<WithStyles(Grid)
  container={true}
  direction="column"
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Typography)
      variant="h6"
    >
      Регистрация
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <ReduxForm />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Button)
      onClick={[MockFunction]}
      variant="contained"
    >
      Зарегистрироваться
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      Назад
    </WithStyles(Button)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
